ADAL/src/ADEnrollmentGateway.m

```diff
@@ -110,71 +125,120 @@ + (NSString *)allEnrollmentIds
 }
 
-+ (NSString *)enrollmentIdForUserId:(NSString *)userId error:(NSError**) error
++ (NSString *)allIntuneMAMResourcesJSON
+{
+    @synchronized(self)
+    {
+        if (s_intuneResourceJSON)
+            return s_intuneResourceJSON;
+    }
+    
+    return [[NSUserDefaults standardUserDefaults] objectForKey:AD_INTUNE_RESOURCE_ID_KEY];
+}
+
++ (NSString *)enrollmentIdForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing *) error
 {
-    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary * dic) {
+    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary *dic) {
         return [[dic objectForKey:USER_ID] isEqualToString:userId];
     }
                                                        error:error];
 }
 
-+ (NSString *)enrollmentIdForUserObjectId:(NSString *)userObjectId tenantId:(NSString *)tenantId error:(NSError**) error
++ (NSString *)enrollmentIdForUserObjectId:(NSString *)userObjectId tenantId:(NSString *)tenantId error:(ADAuthenticationError *__autoreleasing *) error
 {
-    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary * dic) {
+    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary *dic) {
         return [[dic objectForKey:OID] isEqualToString:userObjectId] && [[dic objectForKey:TID] isEqualToString:tenantId];
     }
                                                        error:error];
 }
 
-+ (NSString *)enrollmentIdForUniqueAccountId:(NSString *)uniqueAccountId error:(NSError**) error
++ (NSString *)enrollmentIdForUniqueAccountId:(NSString *)uniqueAccountId error:(ADAuthenticationError *__autoreleasing *) error
 {
-    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary * dic) {
+    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary *dic) {
         return [[dic objectForKey:UNIQUE_ACCOUNT_ID] isEqualToString:uniqueAccountId];
     }
                                                        error:error];
 }
 
-+ (NSString *)intuneMamResource:(NSString *)authority error:(NSError**) error
++ (NSString *)enrollmentIdIfAvailable:(ADAuthenticationError *__autoreleasing *) error
 {
-    NSString* resourceJSON = nil;
-
-    @synchronized(self)
-    {
-        if (s_intuneResourceJSON)
-        {
-            resourceJSON = s_intuneResourceJSON;
-        }
+    // this will just return the first enrollment ID
+    return [ADEnrollmentGateway getEnrollmentIDForIdentifier:^BOOL(NSDictionary * __unused dic) {
+        return true;
     }
+                                                       error:error];
+}
+
++ (NSString *)enrollmentIDForUniqueAccountID:(NSString *) homeUserID userID:(NSString *) userID error:(ADAuthenticationError *__autoreleasing *) error
+{
+    NSString *enrollmentID;
+    enrollmentID = homeUserID ? [ADEnrollmentGateway enrollmentIdForUniqueAccountId:homeUserID error:error] : nil;
```

Comment:
please see my previous comment about logic update for this method (it got collapsed with code style updates, so just making sure to address it)